From 7813ad93bdf61cbfa2a36a5e8233cffde350e825 Mon Sep 17 00:00:00 2001
From: Oliver Schwartz <Oliver.Schwartz@gmx.de>
Date: Fri, 11 Oct 2019 11:13:33 +0200
Subject: [PATCH 1/3] core: Add configuration option for allowed IDs in SMC SiP
 calls.

This patch adds a new configuration parameter, allowed_sip_ids, as an array
of integers to the cell configuration. It specifies the IDs of SMC SiP calls
that are forwarded to the ARM trusted firmware.

Signed-off-by: Oliver Schwartz <Oliver.Schwartz@gmx.de>
---
 include/jailhouse/cell-config.h | 13 +++++++++++--
 tools/jailhouse-cell-linux      |  2 +-
 tools/jailhouse-hardware-check  |  2 +-
 3 files changed, 13 insertions(+), 4 deletions(-)

diff --git a/include/jailhouse/cell-config.h b/include/jailhouse/cell-config.h
index d435b9f..3a61e4b 100644
--- a/include/jailhouse/cell-config.h
+++ b/include/jailhouse/cell-config.h
@@ -50,7 +50,7 @@
  * Incremented on any layout or semantic change of system or cell config.
  * Also update HEADER_REVISION in tools.
  */
-#define JAILHOUSE_CONFIG_REVISION	13
+#define JAILHOUSE_CONFIG_REVISION	14
 
 #define JAILHOUSE_CELL_NAME_MAXLEN	31
 
@@ -98,6 +98,7 @@ struct jailhouse_cell_desc {
 	__u32 num_pci_devices;
 	__u32 num_pci_caps;
 	__u32 num_stream_ids;
+	__u32 num_allowed_sip_ids;
 
 	__u32 vpci_irq_base;
 
@@ -291,7 +292,8 @@ jailhouse_cell_config_size(struct jailhouse_cell_desc *cell)
 		cell->num_pio_regions * sizeof(struct jailhouse_pio) +
 		cell->num_pci_devices * sizeof(struct jailhouse_pci_device) +
 		cell->num_pci_caps * sizeof(struct jailhouse_pci_capability) +
-		cell->num_stream_ids * sizeof(__u32);
+		cell->num_stream_ids * sizeof(__u32) +
+		cell->num_allowed_sip_ids * sizeof(__u32);
 }
 
 static inline __u32
@@ -362,4 +364,11 @@ jailhouse_cell_stream_ids(const struct jailhouse_cell_desc *cell)
 		cell->num_pci_caps * sizeof(struct jailhouse_pci_capability));
 }
 
+static inline const __u32 *
+jailhouse_cell_allowed_sip_ids(const struct jailhouse_cell_desc *cell)
+{
+	return (const __u32 *)((void *)jailhouse_cell_stream_ids(cell) +
+		cell->num_stream_ids * sizeof(__u32));
+}
+
 #endif /* !_JAILHOUSE_CELL_CONFIG_H */
-- 
2.7.4

